extensions/ql-vscode/src/view/variant-analysis/VariantAnalysisRepositoriesStats.tsx

```diff
@@ -0,0 +1,54 @@
+import * as React from 'react';
+import styled from 'styled-components';
+import { VariantAnalysisStatus } from '../../remote-queries/shared/variant-analysis';
+import { formatDecimal } from '../../pure/number';
+
+type Props = {
+  variantAnalysisStatus: VariantAnalysisStatus;
+
+  totalRepositoryCount: number;
+  completedRepositoryCount?: number | undefined;
+
+  queryResult?: 'warning' | 'stopped';
+};
+
+const Icon = styled.span`
```

Comment:
That would be useful, but I'm worried the `margin-left: 0.3em` is not always reusable. I've removed the `margin-left` from the reusable components and [it's now applied from the `Container` component in `VariantAnalysisRepositoriesStats`](https://github.com/github/vscode-codeql/blob/617f7bab0a35f57cce5df086f96a22ddb4d7fe7c/extensions/ql-vscode/src/view/variant-analysis/VariantAnalysisRepositoriesStats.tsx#L8-L10). How it's currently done is the [styled-components way of referring to other components](https://styled-components.com/docs/advanced#referring-to-other-components), but I don't really like that we now need to be worried about the implementation details of the icon.

The alternative would be to wrap it in another styled-component, but that would require another three styled-components. We could also add the `margin-left` in the reusable component, but then we'd need to override/remove it when we want to reuse it somewhere where we don't need the margin.

Reply:
Nice! Thanks for extracting them out. What you've done makes sense - I think the margin shouldn't be applied at the re-usable component level as it makes it potentially less re-usable.

Instead of adding the margin in the `Container`, I believe we could add some `VerticalSpace` next to icons. I know it's a bit controversial, but to me it makes sense to be explicit like that. I'll leave it up to you which approach you go with.